site/content/docs/developer/building-zap-with-eclipse.md

```diff
@@ -0,0 +1,82 @@
+---
+title: "Building ZAP with Eclipse"
+description: "This guide will get you started with building and running ZAP in the Eclipse IDE."
+tags: 
+- guide
+- tutorial
+type: page
+date: "2021-03-18"
+---
+
+If you want to make changes to ZAP using the Eclipse IDE then you are in the right place.
+
+## Preparation
+You will need to have followed the [Quick Start Guide to Building ZAP](../quick-start-build/) and installed a version of [Eclipse](https://www.eclipse.org/) suitable for Java development. We would suggest the "Eclipse IDE for Java Developers" [package](https://www.eclipse.org/downloads/packages/).
+
+## Import the ZAP Repos
+In Eclipse select the 'File' menu, then 'Import...', and then 'Gradle' &#8594; 'Existing Gradle Project':
+
+![Eclipse Gradle Import](/img/docs/developer/eclipse-import.png)
+
+In the following dialog enter the full directory name for one of the ZAP repos you cloned before and click on 'Finish'.
+
+Repeat this process until you have imported all of the ZAP repos.
+
+It is worth noting that some of the zap-extension add-ons may fail to compile if you do not have the required environment for them. This is to be expected and is not a problem unless you want to work on those add-ons, in which case see their READMEs for more details.
+
+## Running ZAP
+In the Eclipse Project Explorer navigate to:
```

Comment:
Alternatively (or additionally) we could suggest using the `zaproxy:run` gradle task?

Reply:
That's counter-productive IMO, if we are using the IDE better make full use of it.

Reply:
+1 but we can also use gradle tasks from within the IDE?

Reply:
Ha! I've added a note saying you can do that. Whats the verdict - include or remove?

Reply:
Yes, that's necessary when installing or deploying the add-ons but running ZAP shouldn't be needed since the IDE allows to run it (and configure it more easily, plus debug).